src/main/java/cache/StationsCache.java

```diff
@@ -0,0 +1,114 @@
+package cache;
+
+import aws.CitiBikeRequest;
+import aws.CitiBikeResponse;
+import com.google.gson.Gson;
+import service.LambdaService;
+import service.LambdaServiceFactory;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.HeadObjectRequest;
+import software.amazon.awssdk.services.s3.model.HeadObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.Duration;
+import java.time.Instant;
+
+public class StationsCache {
+    private static final String BUCKET_NAME = "levy-city";
+    private static final String KEY_NAME = "station_information.json";
+    private CitiBikeResponse stations;
+    private Instant lastModified;
+    private final S3Client s3Client;
+    private final LambdaService service;
+
+    public StationsCache() {
+        this.s3Client = S3Client.builder().region(Region.US_EAST_1).build();
+        this.service = new LambdaServiceFactory().getService();
```

Comment:
This code connects to your lambda.

The StationsCache is supposed to be IN your lambda